src/agimus_pytroller.cpp

```diff
@@ -276,9 +300,32 @@ controller_interface::CallbackReturn AgimusPytroller::on_deactivate(
   return controller_interface::CallbackReturn::SUCCESS;
 }
 
+std::vector<hardware_interface::CommandInterface>
+AgimusPytroller::on_export_reference_interfaces() {
+  std::vector<hardware_interface::CommandInterface> reference_interfaces;
+
+  const auto names = params_.reference_interfaces;
+  reference_interfaces_.resize(names.size());
+  for (std::size_t i = 0; i < names.size(); i++) {
+    reference_interfaces.push_back(hardware_interface::CommandInterface(
+        get_node()->get_name(), names[i], &reference_interfaces_[i]));
+    // walkaround as CommandInterface doesn't have the copy constructor
+    command_reference_interfaces_.push_back(
+        hardware_interface::CommandInterface(get_node()->get_name(), names[i],
+                                             &reference_interfaces_[i]));
+  }
+
+  return reference_interfaces;
+}
+
 controller_interface::return_type
-AgimusPytroller::update(const rclcpp::Time &time,
-                        const rclcpp::Duration &period) {
+AgimusPytroller::update_reference_from_subscribers() {
```

Comment:
maybe you want to take a look at the LFC code to make it compatible Jazzy/Humble?